### PR TITLE
[Spaces] Local Storage

### PR DESCRIPTION
**Pull request by abelgardep**
## Related Issues
Implements https://github.com/owncloud/android/issues/3808

Library PR (if needed): https://github.com/owncloud/android-library/pull/537
_____

### CODE REVIEW

File: owncloudData/src/main/java/com/owncloud/android/data/storage/LocalStorageProvider.kt [hunk 1]
```diff
@@ -58,9 +58,16 @@ sealed class LocalStorageProvider(private val rootFolderName: String) {
      * file.
      */
     fun getDefaultSavePathFor(
-        accountName: String?,
-        remotePath: String
-    ): String = getAccountDirectoryPath(accountName) + remotePath
+        accountName: String,
+        remotePath: String,
+        spaceId: String?,
```
Comment (JuancaG05) [inline owncloudData/src/main/java/com/owncloud/android/data/storage/LocalStorageProvider.kt:63]: Maybe setting this parameter with default value `null`? Just like in `getTemporalPath`

**Review by JuancaG05 — APPROVED** [review]
Great! Just a tiny comment, but from my point of view, ready to go 🚀

File: owncloudData/src/main/java/com/owncloud/android/data/storage/LocalStorageProvider.kt (hunk 1 above)
Comment (abelgardep) [reply owncloudData/src/main/java/com/owncloud/android/data/storage/LocalStorageProvider.kt:63]: No default value is needed for this one. We want to make sure that we receive the spaceId, even if it's null.

I would change the one from `getTemporalPath` though

**Comment by jesmrec** [comment]
Approved from my side

**Pull request merged**